lib/uplink/buckets.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package uplink
+
+import (
+	"context"
+	"errors"
+
+	"storj.io/storj/pkg/storj"
+)
+
+// BucketOpts holds the cipher, path, key, and enc. scheme for each bucket since they
+// can be different for each
+type BucketOpts struct {
+	PathCipher       storj.Cipher
+	EncPathPrefix    storj.Path
+	Key              storj.Key
+	EncryptionScheme storj.EncryptionScheme
+}
+
+// CreateBucketOptions holds the bucket opts
+type CreateBucketOptions struct {
+	PathCipher storj.Cipher
+	// this differs from storj.CreateBucket's choice of just using storj.Bucket
+	// by not having 2/3 unsettable fields.
```

Comment:
I might be way out of date, but I believe CreateBucket does need to know the root encryption key and encryption scheme, so it can create the toplevel bucket metainfo in pointerdb.

Reply:
Or no, just the encryption scheme.

Reply:
This part of the code is incomplete, so I'll add a TODO here for that as well.

Reply:
Okay, I added `EncConfig` struct to this so it can be made with different encryption